### PR TITLE
Retry synchronous commit after processing a message.

### PR DESCRIPTION
**Pull request by chasdevs**
Attempts an application-level commit retry in addition to setting the `retries` librdkafka config to 3.

### CODE REVIEW

File: KafkaPhp/Consumer/Consumer.php [hunk 1]
```diff
@@ -148,7 +152,15 @@ private function poll(): void
                     $this->logger->debug('Processing message.', ['offset' => $message->offset]);
                     $record = $this->serializer->deserialize($message->payload);
                     $this->recordProcessor->process($record);
-                    $this->kafkaClient->commit($message);
+                    try {
+                        $this->kafkaClient->commit($message);
+                    } catch (Exception $rdkafkaException) {
+                        if (preg_match("/Request timed out/", $rdkafkaException->getMessage())) {
+                            // retry once on timeout
+                            $this->logger->warning("Retrying commit after timeout.", ["message" => $message]);
```
Comment (balexander) [inline KafkaPhp/Consumer/Consumer.php]: 👍

File: KafkaPhp/Consumer/ConsumerBuilder.php [hunk 2]
```diff
@@ -55,6 +55,7 @@ public function __construct(
         $this->groupId = $groupId;
         $this->config->set(ConsumerConfigOptions::GROUP_ID, $this->groupId);
         $this->config->set(ConsumerConfigOptions::AUTO_OFFSET_RESET, $this->offsetReset);
+        $this->config->set(ConfigOptions::RETRIES, 3);
```
Comment (balexander) [inline KafkaPhp/Consumer/ConsumerBuilder.php]: There's a `DEFAULT_RETRIES` if you're so inclined

Comment (chasdevs) [reply KafkaPhp/Consumer/ConsumerBuilder.php]: We have a few different definitions of "retries" at this point. Think we should name them more specifically?

Comment (balexander) [reply KafkaPhp/Consumer/ConsumerBuilder.php]: @chasdevs Yeah we should... but naming is hard

**Pull request merged**